CivilServiceUSA/us-states
# Updates: Mississippi Flag; Node.Js FileSystem Module

#### What's this PR do?
- [ ] The Mississippi state flag changed on July 1st, 2020.  I have updated the large and small versions of the flag based on the the image appearing on [the official Mississippi Department of Archives and History history site](https://www.mdah.ms.gov/news/state-flag-commission-picks-new-magnolia-flag-november-ballot) .  

- [ ] Updates to node.js have occurred within the File Server module (fs) that required some minor updates to the converter.js files.

#### Where should the reviewer start?

In regards to the Mississippi flag, only two images were updated, review the two modified files in the pull request.  Note that once this pull request has been approved, the CDN will also need to be updated.

The updates to the converter scripts affected the ordering of the data/states.sql file, but none of the data has changed. 

#### How should this be manually tested?

Compare the /images/flags/mississippi-small.png and /images/flags/mississippi-large.png files, sizes, widths and heights to the previous versions.  This can be viewed within the pull request file comparison.

For the node.js update, just try running the commands as follows and ensure no error occurs and the YAML, SQL, CSV and XML files are generated.

```
npm install
npm run convert
```

Initially, after executing the convert request, the following error was occurring:

> node:internal/validators:224
> throw new ERR_INVALID_CALLBACK(callback);
> ^
> 
> TypeError [ERR_INVALID_CALLBACK]: Callback must be a function. Received undefined
> at maybeCallback (node:fs:177:3)
> at Object.writeFile (node:fs:2118:14)
> at Object. (/Projects/opensource/us-states/converters/json-to-xml.js:7:4)
> at Module._compile (node:internal/modules/cjs/loader:1105:14)
> at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
> at Module.load (node:internal/modules/cjs/loader:981:32)
> at Function.Module._load (node:internal/modules/cjs/loader:822:12)
> at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
> at node:internal/main/run_main_module:17:47 {
> code: 'ERR_INVALID_CALLBACK'
> }
> 

#### Any background context you want to provide?

NOTE: This is my very first open source contribution, so any constructive feedback is welcome.  Thanks!

Here is a reference on [the official Mississippi Department of Archives and History history site](https://www.mdah.ms.gov/news/state-flag-commission-picks-new-magnolia-flag-november-ballot) with the same image and background information on the updated flag.

#### What are the relevant github issues?
#2 
#4  
#3 

Two and four are related, I just wrote it up in the requested templated format.

#### Screenshots (if appropriate)
![mississippi-small](https://user-images.githubusercontent.com/6627582/204190713-ea4aa11d-c643-41ae-b30b-2751f5cabaee.png)

![mississippi-large](https://user-images.githubusercontent.com/6627582/204190723-ca600a64-0bf6-4ff0-a080-d095dcea4b15.png)

#### What gif best describes this PR or how it makes you feel?

![Alt Text](https://media.giphy.com/media/HzPjSwiIDf8OaMrozi/giphy.gif)

#### Definition of Done:

- [x] You have added code comments to all code being submitted (GitHub Commit Messages)
- [x] You have updated the README file (if appropriate) (I did not update the README)
